steps/s3/session.go

```diff
@@ -0,0 +1,184 @@
+	logrus.Debugf("Creating a new bucket: %s", bucket)
+	cparams := &s3.CreateBucketInput{
+		Bucket: aws.String(bucket),
+	}
+
+	s3Client := s3.New(s.Client)
+	_, err := s3Client.CreateBucket(cparams)
+	if err != nil {
+		logrus.Errorf("Error creating a new bucket: %s, err: %v", bucket, err)
+	}
+	return nil
+}
+
+// validateS3File checks the existence of a file in S3.
+func (s *Session) validateS3File(ctx context.Context, bucket, key string) error {
+	logger := GetLogger()
+	logger.LogOperation("validate", bucket, key)
+	s3svc := s3.New(s.Client)
+	exists, err := s.s3KeyExists(s3svc, bucket, key)
+	if err != nil {
+		return err
+	}
+	if !exists {
+		return fmt.Errorf("failed validating s3 file exits: no file exists in bucket '%s' with name '%s' ", bucket, key)
+	}
+	return nil
+}
+
+// validateS3FileWithContent checks the existence of a file in S3 with the content specified.
+func (s *Session) validateS3FileWithContent(ctx context.Context, bucket, key, message string) error {
+	expected := strings.TrimSpace(message)
+	logger := GetLogger()
+	logger.LogOperation("validate", bucket, key)
+	downloader := s3manager.NewDownloader(s.Client)
+	buf := aws.NewWriteAtBuffer([]byte{})
+	_, err := downloader.Download(buf, &s3.GetObjectInput{
+		Bucket: aws.String(bucket),
+		Key:    aws.String(key),
+	})
+	if err != nil {
+		return fmt.Errorf("unable to upload %q to %q, %v", key, bucket, err)
+	}
+	actual := strings.TrimSpace(string(buf.Bytes()))
+	if expected != actual {
+		return fmt.Errorf("failed validating s3 bucket '%s' file '%s' content: expected:\n%s\n\nactual:\n%s", bucket, key, expected, actual)
+	}
+	return nil
+}
+
+// deleteS3File deletes the file in S3.
+func (s *Session) deleteS3File(ctx context.Context, bucket, key string) error {
+	logger := GetLogger()
+	logger.LogOperation("delete", bucket, key)
+	s3svc := s3.New(s.Client)
+	input := &s3.DeleteObjectInput{
+		Bucket: aws.String(bucket),
+		Key:    aws.String(key),
+	}
+	_, err := s3svc.DeleteObject(input)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// s3KeyExists checks the existence of a key in a S3 bucket.
+func (s *Session) s3KeyExists(s3svc *s3.S3, bucket string, key string) (bool, error) {
+	_, err := s3svc.HeadObject(&s3.HeadObjectInput{
+		Bucket: aws.String(bucket),
+		Key:    aws.String(key),
+	})
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case "NotFound": // s3.ErrCodeNoSuchKey does not work, aws is missing this error code so we hardwire a string
+				return false, nil
+			default:
+				return false, err
+			}
+		}
+		return false, err
+	}
+	return true, nil
+}
+
+// CleanUp cleans session by deleting all documents created in S3
+func (s *Session) CleanUp(ctx context.Context) {
+	for _, file := range s.CreatedDocuments {
+		err := s.deleteS3File(ctx, file.bucket, file.key)
+		if err != nil {
+			logrus.Errorf("Failure on deletion of s3 file '%s' in bucket '%s', err %v", file.key, file.bucket, err)
```

Comment:
You forgot to change this logrus by its logger:=GetLogger()

Reply:
Done